### PR TITLE
Enhancement/2428 source link

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue #2428 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics/datastore/service.js [hunk 1]
```diff
@@ -91,7 +87,12 @@ export const selectors = {
 		}
 
 		const argsSegment = reportArgsToURLSegment( reportArgs );
-		const path = joinPaths( '/report', type, `a${ accountID }w${ internalWebPropertyID }p${ profileID }`, argsSegment, '/' );
+		let path = escapeURI`/report/${ type }/a${ accountID }w${ internalWebPropertyID }p${ profileID }`;
+
+		if ( argsSegment ) {
+			path += escapeURI`/${ argsSegment }`;
+		}
+		path += '/';
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/datastore/service.js:95]:
```suggestion
		let path = escapeURI`/report/${ type }/a${ accountID }w${ internalWebPropertyID }p${ profileID }/`;
		if ( argsSegment ) {
			path += escapeURI`${ argsSegment }/`;
		}
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
@asvinb overall it looks good to me. Just added one suggestion and it's ready to go.

**Comment by eugene-manuilov** [comment]
@asvinb could you please take a look at js tests too? Looks like it has stopped working after merging it with the develop branch.

File: assets/js/modules/analytics/datastore/service.js (hunk 1 above)
Comment (asvinb) [reply assets/js/modules/analytics/datastore/service.js:95]: nice 👍

**Review by eugene-manuilov — APPROVED** [review]
:+1:

**Review by felixarntz — APPROVED** [review]
Great!

**Pull request merged**